### PR TITLE
Add documentation to native metrics

### PR DESCRIPTION
**Pull request by cuent**
## What does this PR do?

Add documentation for native metrics

### CODE REVIEW

File: pytorch_lightning/metrics/functional/classification.py [hunk 1]
```diff
@@ -168,13 +252,61 @@ def precision_recall(pred: torch.Tensor, target: torch.Tensor,
...
+           Available reduction methods:
+
+           - elementwise_mean: takes the mean
+
+           - none: pass array
+
+           - sum: add elements
+
+    Output:
+        Tensor with recall.
```
Comment (Borda) [inline pytorch_lightning/metrics/functional/classification.py:308]:
```suggestion
    Args:
        pred: estimated probabilities
        target: ground-truth labels
        num_classes: number of classes
        reduction: method for reducing recall values (default: takes the mean)
           Available reduction methods:

           - elementwise_mean: takes the mean
           - none: pass array
           - sum: add elements

    Return:
        Tensor with recall.
```

File: pytorch_lightning/metrics/functional/classification.py [hunk 2]
```diff
@@ -168,13 +252,61 @@ def precision_recall(pred: torch.Tensor, target: torch.Tensor,
...
+           Available reduction methods:
+
+           - elementwise_mean: takes the mean
+
+           - none: pass array
+
+           - sum: add elements
+
+    Output:
+        Tensor with precision.
```
Comment (Borda) [inline pytorch_lightning/metrics/functional/classification.py:277]:
```suggestion
    Args:
        pred: estimated probabilities
        target: ground-truth labels
        num_classes: number of classes
        reduction: method for reducing precision values (default: takes the mean)
           Available reduction methods:

           - elementwise_mean: takes the mean
           - none: pass array
           - sum: add elements

    Return:
        Tensor with precision.
```

File: pytorch_lightning/metrics/functional/classification.py [hunk 3]
```diff
@@ -149,6 +209,30 @@ def precision_recall(pred: torch.Tensor, target: torch.Tensor,
...
+           Available reduction methods:
+
+           - elementwise_mean: takes the mean
+
+           - none: pass array
+
+           - sum: add elements
+
+    Output:
+        Tensor with precision and recall
```
Comment (Borda) [inline pytorch_lightning/metrics/functional/classification.py:234]:
```suggestion
    Args:
        pred: estimated probabilities
        target: ground-truth labels
        num_classes: number of classes
        reduction: method for reducing precision-recall values (default: takes the mean)
           Available reduction methods:

           - elementwise_mean: takes the mean
           - none: pass array
           - sum: add elements

    Return:
        Tensor with precision and recall
```

File: pytorch_lightning/metrics/functional/classification.py [hunk 4]
```diff
@@ -129,6 +175,20 @@ def accuracy(pred: torch.Tensor, target: torch.Tensor,
...
+
+    Args:
+        pred: estimated targets
+
+        target: groud truth labels
+
+        normalize: normalizes confusion matrix
+
+    Output:
+        Tensor, confusion matrix C [num_classes, num_classes ]
```
Comment (Borda) [inline pytorch_lightning/metrics/functional/classification.py:190]:
```suggestion
    Args:
        pred: estimated targets
        target: ground truth labels
        normalize: normalizes confusion matrix

    Return:
        Tensor, confusion matrix C [num_classes, num_classes ]
```

File: pytorch_lightning/metrics/functional/classification.py [hunk 5]
```diff
@@ -116,6 +139,29 @@ def stat_scores_multiple_classes(pred: torch.Tensor, target: torch.Tensor,
...
+           Available reduction methods:
+
+           - elementwise_mean: takes the mean
+
+           - none: pass array
+
+           - sum: add elements
+
+    Output:
+         A Tensor with the classification score.
```
Comment (Borda) [inline pytorch_lightning/metrics/functional/classification.py:163]:
```suggestion
    Args:
        pred: predicted labels
        target: ground truth labels
        num_classes: number of classes
        reduction: a method for reducing accuracies over labels (default: takes the mean)
           Available reduction methods:

           - elementwise_mean: takes the mean
           - none: pass array
           - sum: add elements

    Return:
         A Tensor with the classification score.
```

File: pytorch_lightning/metrics/functional/classification.py [hunk 6]
```diff
@@ -94,6 +114,9 @@ def stat_scores_multiple_classes(pred: torch.Tensor, target: torch.Tensor,
         argmax_dim: if pred is a tensor of probabilities, this indicates the
             axis the argmax transformation will be applied over
 
+    Output:
```
Comment (Borda) [inline pytorch_lightning/metrics/functional/classification.py:117]:
```suggestion
    Return:
```

File: pytorch_lightning/metrics/functional/classification.py [hunk 7]
```diff
@@ -63,6 +80,9 @@ def stat_scores(pred: torch.Tensor, target: torch.Tensor,
...
```
Comment (Borda) [inline pytorch_lightning/metrics/functional/classification.py:83]:
```suggestion
    Return:
```

File: pytorch_lightning/metrics/functional/classification.py [hunk 8]
```diff
@@ -29,14 +29,31 @@ def to_onehot(tensor: torch.Tensor,
...
     """ Returns the number of classes for a given prediction and
-        target tensor
+        target tensor,
+
+        Args:
+            pred: predicted values
+            target: true labels
+            num_classes: number of classes if known (default: None)
+
+        Output:
```
Comment (Borda) [inline pytorch_lightning/metrics/functional/classification.py:55]:
```suggestion
        Return:
```

File: pytorch_lightning/metrics/functional/classification.py [hunk 9]
```diff
@@ -29,14 +29,31 @@ def to_onehot(tensor: torch.Tensor,
...
-    """ Converts a tensor of probabilities to a dense label tensor """
+    """ Converts a tensor of probabilities to a dense label tensor
+
+    Args:
+        tensor: probabilities to get the categorical label [N, d1, d2, ...]
+
+        argmax_dim: dimension to apply (default: 1)
+
+    Output:
+        A tensor with categorical labels [N, d2, ...]
```
Comment (Borda) [inline pytorch_lightning/metrics/functional/classification.py:40]:
```suggestion
    Args:
        tensor: probabilities to get the categorical label [N, d1, d2, ...]
        argmax_dim: dimension to apply (default: 1)

    Return:
        A tensor with categorical labels [N, d2, ...]
```

**Comment by Borda** [comment]
@cuent I think it looks good, will you add some more or shall I merge it?

**Comment by cuent** [comment]
I will finish the remaining functions and take a look to the dice score. I will push changes within this weekend 😄

**Comment by Borda** [comment]
> I will finish the remaining functions and take a look to the dice score. I will push changes within this weekend

if I could ask, I would prefer to just finish the docs by Friday so we can release it in 0.8 and then we can add Dice in separate PR to be released in 0.8.1... what do you think?

**Comment by cuent** [comment]
yeah! then I will push later tonight :)

File: pytorch_lightning/metrics/functional/classification.py [hunk 10]
```diff
@@ -116,6 +138,23 @@ def stat_scores_multiple_classes(pred: torch.Tensor, target: torch.Tensor,
 def accuracy(pred: torch.Tensor, target: torch.Tensor,
              num_classes: Optional[int] = None,
              reduction='elementwise_mean') -> torch.Tensor:
+    '''
```
Comment (awaelchli) [inline pytorch_lightning/metrics/functional/classification.py:141]:
```suggestion
    """
```
for consistency with the rest of the docs

**Comment by cuent** [comment]
it should be good to go, lmk if I can help with something else  😄

**Pull request merged**